korma/src/commonMain/kotlin/com/soywiz/korma/geom/Quaternion.kt

```diff
@@ -2,80 +2,85 @@ package com.soywiz.korma.geom
+
+    inline val length: Float get() = sqrt(quaternionLengthSquared(x, y, z, w))
+    inline val lengthSqr: Float get() = quaternionLengthSquared(x, y, z, w)
+
+    operator fun get(index: Int): Float = when (index) {
         0 -> x
         1 -> y
         2 -> z
         3 -> w
-        else -> Double.NaN
+        else -> Float.NaN
     }
-    inline fun setToFunc(callback: (Int) -> Double) = setTo(callback(0), callback(1), callback(2), callback(3))
-    fun setTo(x: Double, y: Double, z: Double, w: Double): Quaternion {
+    inline fun setToFunc(callback: (Int) -> Float) = setTo(callback(0), callback(1), callback(2), callback(3))
+
+    fun setTo(x: Float, y: Float, z: Float, w: Float): Quaternion {
         this.x = x
         this.y = y
         this.z = z
         this.w = w
         return this
     }
-    fun setTo(x: Int, y: Int, z: Int, w: Int): Quaternion = setTo(x.toDouble(), y.toDouble(), z.toDouble(), w.toDouble())
-    fun setTo(x: Float, y: Float, z: Float, w: Float): Quaternion = setTo(x.toDouble(), y.toDouble(), z.toDouble(), w.toDouble())
-    fun setTo(euler: EulerRotation): Quaternion = EulerRotation.toQuaternion(euler, this)
-    fun setTo(other: Quaternion): Quaternion = setTo(other.x, other.y, other.z, other.w)
+
+    fun setTo(euler: EulerRotation) = setTo(EulerRotation.toQuaternion(euler, this))
+    fun setTo(other: Quaternion) = setTo(other.x, other.y, other.z, other.w)
 
     fun setEuler(x: Angle, y: Angle, z: Angle): Quaternion = EulerRotation.toQuaternion(x, y, z, this)
     fun setEuler(euler: EulerRotation): Quaternion = EulerRotation.toQuaternion(euler, this)
 
-    fun copyFrom(other: Quaternion): Quaternion = this.setTo(other)
-
     operator fun unaryMinus(): Quaternion = Quaternion(-x, -y, -z, -w)
     operator fun plus(other: Quaternion): Quaternion = Quaternion(x + other.x, y + other.y, z + other.z, w + other.w)
     operator fun minus(other: Quaternion): Quaternion = Quaternion(x - other.x, y - other.y, z - other.z, w - other.w)
-    operator fun times(scale: Double): Quaternion = Quaternion(x * scale, y * scale, z * scale, w * scale)
+    operator fun times(scale: Float): Quaternion = Quaternion(x * scale, y * scale, z * scale, w * scale)
 
     fun negate() = this.setTo(-x, -y, -z, -w)
 
-    inline fun setToFunc(l: Quaternion, r: Quaternion, func: (l: Double, r: Double) -> Double) = setTo(
+    inline fun setToFunc(l: Quaternion, r: Quaternion, func: (l: Float, r: Float) -> Float) = setTo(
         func(l.x, r.x),
         func(l.y, r.y),
         func(l.z, r.z),
         func(l.w, r.w)
     )
-    fun setToSlerp(left: Quaternion, right: Quaternion, t: Double, tleft: Quaternion = Quaternion(), tright: Quaternion = Quaternion()): Quaternion {
-        val tleft = tleft.copyFrom(left).normalize()
-        val tright = tright.copyFrom(right).normalize()
+
+    fun setToSlerp(left: Quaternion, right: Quaternion, t: Float, tleft: Quaternion = Quaternion(), tright: Quaternion = Quaternion()) {
```

Comment:
Did we vote in the end removing chainable calls? If not, we can make a RFC now and vote